src/argus_htmx/themes/views.py

```diff
@@ -40,9 +40,7 @@ def theme_names(request: HtmxHttpRequest) -> HttpResponse:
 
 @require_POST
 def change_theme(request: HtmxHttpRequest) -> HttpResponse:
-    theme = get_preference(request, "argus_htmx", "theme")
     success = save_preference(request, request.POST, "argus_htmx", "theme")
     if success:
-        theme = get_preference(request, "argus_htmx", "theme")
-        return HttpResponse(theme)
```

Comment:
How is this a regression? change_theme has always returned the new theme name on success, the theme dropdown does not collapse after changing.

Reply:
Because we are appending the messages to the response we need to be sure that the response is a valid partial html. I therefore wrapped the theme in a span using the `_current_theme.html ` template

Reply:
Ah, thx